metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/Taxonomy.java

```diff
@@ -54,11 +56,33 @@
     @NotNull
     @JsonProperty
     @Size(max = 255, min = 1)
+    @Column(unique = true)
     private String name;
 
-    @ManyToMany
+    @ApiModelProperty(position = 4)
     @JsonProperty
-    private List<Taxonomy> ancestors;
+    @Column
+    private String rank;
```

Comment:
I think that we should still have a `@NotNull` condition on the `rank` property. When NCBI specifies that a taxon is of "no rank", this doesn't really have any special meaning which we would need to process; we can just treat all ranks equally and, in this case, simply store the "no rank" literal in this property. However, it must always be specified to contain _something,_ hence we need the `@NotNull` annotation